File: AspNetMvcUnleashed/0672329980_source/Chapter03/VB/MvcApplication1/MvcApplication1/Controllers/WidgetController.vb
Public Class WidgetController
    Inherits System.Web.Mvc.Controller

    Function Index() As ActionResult
        Return View()
    End Function


    Function Details(ByVal id As Integer?) As ActionResult
        If Not id.HasValue Then
            Return RedirectToAction("Index")
        End If

        Return View()
    End Function


End Class

File: AspNetMvcUnleashed/0672329980_source/Chapter03/CS/MvcApplication1/MvcApplication1/Controllers/WidgetController.cs
using System.Web.Mvc;

namespace MvcApplication1.Controllers
{
    public class WidgetController : Controller
    {
        //
        // GET: /Widget/

        public ActionResult Index()
        {
            return View();
        }


        //
        // POST: /Widget/Create

        public ActionResult Details(int? id)
        {
            if (!id.HasValue)
                return RedirectToAction("Index", "ProductController");

            return View();
        }



    }
}
